firedrake/slate/slac/utils.py

```diff
@@ -0,0 +1,130 @@
+from __future__ import absolute_import, print_function, division
+
+from coffee import base as ast
+from coffee.visitor import Visitor
+
+from ufl.algorithms.multifunction import MultiFunction
+
+
+class RemoveRestrictions(MultiFunction):
+    """UFL MultiFunction for removing any restrictions on the integrals of forms."""
+    expr = MultiFunction.reuse_if_untouched
+
+    def positive_restricted(self, o):
+        return self(o.ufl_operands[0])
+
+
+class SymbolBho(ast.Symbol):
```

Comment:
It's italian.  How about `SymbolWithFuncallIndexing`?  This is only used internally, so may as well make the name verbose, but descriptive...